### ISSUE TITLE
feat: trainer abstraction

### ISSUE THREAD

**Issue by arilotter** [PsycheFoundation/psyche#75]
Psyche should work with multiple training backends.

We currently only support `tch-rs`, but we'd like to support native Python trainers e.g. [torchtitan](https://github.com/pytorch/torchtitan) to take advantage of their optimizations.

There's no reason we can't also support arbitrary frameworks, too, as long as we have the ability to shuttle some sort of Tensor in & out.

This opens the door for integrating frameworks like [burn](https://github.com/tracel-ai/burn).
